aiosparql/client.py

```diff
@@ -80,19 +79,30 @@ def format_field(self, value, format_spec):
 
 
 class SPARQLClient:
-    def __init__(self,
-                 endpoint: Optional[str] = None,
+    def __init__(self, endpoint: str, *,
                  update_endpoint: Optional[str] = None,
-                 prefixes: Optional[List[Namespace]] = None,
-                 graph: Optional[str] = None,
+                 prefixes: Optional[Dict[str, IRI]] = None,
+                 graph: Optional[IRI] = None,
                  **kwargs):
         self._closed = False
-        self.endpoint = endpoint or ENV['MU_SPARQL_ENDPOINT']
-        self.graph = graph or IRI(ENV['MU_APPLICATION_GRAPH'])
-        self.update_endpoint = update_endpoint or self.endpoint
+        self._endpoint = endpoint
+        self._update_endpoint = update_endpoint
+        self._graph = graph
         self.session = aiohttp.ClientSession(**kwargs)
         self._generate_prefixes(prefixes)
 
+    @property
+    def endpoint(self):
+        return self._endpoint
+
+    @property
+    def update_endpoint(self):
```

Comment:
On the specs they just say:

> A service (often referred to by the informal term SPARQL endpoint) that accepts and processes update requests is referred to as an update service.

https://www.w3.org/TR/sparql11-update/#graphStoreQueryServices